internal/deviations/deviations.go

```diff
@@ -277,8 +277,13 @@ func SwVersionUnsupported(_ *ondatra.DUTDevice) bool {
 }
 
 // HierarchicalWeightResolutionTolerance returns the allowed tolerance for BGP traffic flow while comparing for pass or fail conditions.
-func HierarchicalWeightResolutionTolerance(_ *ondatra.DUTDevice) float64 {
-	return *hierarchicalWeightResolutionTolerance
+// Default minimum value is 0.2. Anything less than 0.2 will be set to 0.2.
+func HierarchicalWeightResolutionTolerance(dut *ondatra.DUTDevice) float64 {
+	logErrorIfFlagSet("deviation_hierarchical_weight_resolution_tolerance")
```

Comment:
I think we need to avoid referencing deviation fields directly, because that will cause a nil deference if there are no deviations for the DUT. We should also only call `lookupDUTDeviations(dut)` once in any accessor. Maybe:

```
logErrorIfFlagSet("deviation_hierarchical_weight_resolution_tolerance")
hwrt := lookupDUTDeviations(dut).GetHierarchicalWeightResolutionTolerance()
if minHWRT := 0.2; hwrt < minHWRT {
	return minHWRT
}
return hwrt
```